### PR TITLE
Use one-based var names in Types

### PR DESCRIPTION
**Pull request by eksperimental**
In the same fashion as generated arguments 30373382ec2183d316f14b6ec6a99225855c6d8d

### CODE REVIEW

File: lib/elixir/lib/module/types/unify.ex [hunk 1]
```diff
@@ -830,8 +830,8 @@ defmodule Module.Types.Unify do
     inspect(literal)
   end
 
-  def format_type({:var, index}, _simplify?) do
-    "var#{index}"
+  def format_type({:var, index}, _simplify?) when is_integer(index) do
```
Comment (eksperimental) [inline lib/elixir/lib/module/types/unify.ex:833]: I am not sure if there is a case where the second element of the tuple will not be an index.

Comment (ericmj) [reply lib/elixir/lib/module/types/unify.ex:833]: It will always be an integer.

```suggestion
  def format_type({:var, index}, _simplify?) do
```

**Review by ericmj — APPROVED** [review]
In the future I want letter based type names: `a`, `b`, `c`, ... But that will be affected by user types so I have held off on it. This is a good intermediary change.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by eksperimental** [comment]
Thank you @ericmj  and @josevalim

**Pull request merged**